pkg/pillar/cmd/zedmanager/updatestatus.go

```diff
@@ -412,17 +434,37 @@ func doActivate(ctx *zedmanagerContext, uuidStr string,
 		need := uint64(config.FixedResources.Memory) << 10
 		if remaining < need {
 			var errStr string
+			retryable := false
+			var entities []*types.ErrorEntity
+			retryCondition := "Retry will be triggered when apps will shutdown: "
 			if remaining+halting < need {
 				errStr = fmt.Sprintf("Remaining memory bytes %d app instance needs %d",
 					remaining, need)
 			} else {
 				errStr = fmt.Sprintf("App instance needs %d bytes but only have %d; waiting for halting app instances to free up %d bytes",
 					need, remaining, halting)
+				for _, st := range ctx.pubAppInstanceStatus.GetAll() {
+					status := st.(types.AppInstanceStatus)
+					if status.Activated || status.ActivateInprogress {
+						entities = append(entities, &types.ErrorEntity{EntityID: status.UUIDandVersion.UUID.String(), EntityType: types.ErrorEntityAppInstance})
+						retryCondition = fmt.Sprintf("%s %s;", retryCondition, status.DisplayName)
+						retryable = true
+					}
+				}
```

Comment:
I think this code can be common - use whether or not there are some halting apps. Because this app still depends on some of those apps to shut down.

However, in the case of halting we can restrict the set of AppInstances which are listed to those in HALTING state. Otherwise we list all.